### ISSUE TITLE
[BUG] clang-12: error: clang frontend command failed with exit code 139

### ISSUE THREAD

**Issue by xloem** [android/ndk#1645]
#### Description

While building, the compiler crashed and informed me to report a bug. I have attached its generated data, which appears to likely form a test case.

[clang.log](https://github.com/android/ndk/files/7817765/clang.log)
[MaxUnpoolKernel_cpp_sh.tar.gz](https://github.com/android/ndk/files/7817774/MaxUnpoolKernel_cpp_sh.tar.gz)

Update: This crash went away for me when I switched to ndk 24-beta2.

#### Environment Details

* NDK Version: 23.1.7779620
* Build system: cmake ninja
* Host OS: Linux Ubuntu 20.04.3, running from docker image termux/package-builder
* ABI: arm64-v8a
* NDK API level: 23
* Device API level: n/a

**Comment by pirama-arumuga-nainar** [comment]
This need not be triaged for r24 as mentioned in the original report.

Crashes in r23 clang: SHA `8456c3a789285079ad35d146e487436b5a27b027`
Fixed in r24 clang: SHA `0fe7531b75e54909a7cb7f7080634ab2053964ad`

Next step is to identify a fix using `git bisect` and cherry-pick to the LTS compiler branch.

**Comment by pirama-arumuga-nainar** [comment]
Fix is:

```
commit 7654bb6303d290b19cad29137be810e69a0bf917
Author: Alexey Bataev <a.bataev@outlook.com>
Date:   Wed Mar 24 09:24:24 2021 -0700

    [OPENMP]Fix PR48571: critical/master in outlined contexts cause crash.
    
    If emit inlined region for master/critical directives, no need to clear
    lambda/block context data, otherwise the variables cannot be found and
    it causes a crash at compile time.
    
    Differential Revision: https://reviews.llvm.org/D99280
```

**Comment by pirama-arumuga-nainar** [comment]
https://android-review.googlesource.com/c/toolchain/llvm_android/+/1939337 cherry-picks it to the necessary branch.  Just need a prebuilt update before r23c.

**Comment by DanAlbert** [comment]
> This need not be triaged for r24 as mentioned in the original report.

Whoops, a little overzealous while batch triaging :) Thanks

**Comment by DanAlbert** [comment]
Should be fixed in r23 [build 8486889](https://ci.android.com/builds/branches/aosp-ndk-release-r23/grid?head=8486889&tail=8486889).
